### PR TITLE
feat: support multiple WebAuthn RP origins (rp_origins)

### LINKED ISSUE #143: Permit configuration of multiple RP origins
## Summary
Enable configuration of multiple URLs/URIs as RP origin for WebAuthn.

## Problem statement
To specify an allowed origin for WebAuthn login/registration, the configuration option ".server.rp_origin" is utilized.

This typically contain the frontend's origin ("https://id.siros.org", for example), but [an additional value may be required for usage by native apps](https://github.com/sirosfoundation/wallet-android-wrapper/issues/1).

Since this option is a single string, it is not possible to specify multiple allowed origins.

## Proposed solution
Change the setting to be a list of strings or pass the existing ".server.cors.allowed_origins" setting to the [WebAuth validation function](https://github.com/sirosfoundation/go-wallet-backend/blob/6da42157c849410dad9e0687555dd9b9c1f3d74d/internal/service/webauthn.go#L64).

### PR DESCRIPTION
**Pull request by leifj**
## Summary

Adds a `rp_origins` list field to `ServerConfig` alongside the existing `rp_origin` string field, enabling native app wrappers (Android, iOS) to register and authenticate against a wallet backend that also serves a web frontend.

Closes #143

### `pkg/config/config.go`
- New `RPOrigins []string` field (`yaml:"rp_origins"`, `envconfig:"RP_ORIGINS"`)
- New `GetRPOrigins() []string` helper on `*ServerConfig`:
  - Prepends `RPOrigin` (legacy field) if non-empty
  - Deduplicates the combined list
  - Returns `nil` when both are empty
- `Validate()`: accepts either `rp_origin` or `rp_origins` being non-empty; both empty is still an error
- `defaultConfig()`: explicit `RPOrigins: nil` (no behaviour change)

### `internal/service/webauthn.go`
- `NewWebAuthnServiceWithValidator`: replace `[]string{cfg.Server.RPOrigin}` with `cfg.Server.GetRPOrigins()`

### `pkg/config/config_test.go`
- Updated `TestConfig_Validate_MissingRPOrigin` to cover both fields empty
- Added `TestConfig_Validate_RPOriginsAlone`: validate passes with only `rp_origins` set
- Added `TestServerConfig_GetRPOrigins` (5 table-driven cases): single legacy, list only, legacy prepended, deduplication, both empty

## Backward compatibility

Existing deployments using `rp_origin` require no config change. `GetRPOrigins()` returns `[rp_origin]` exactly as before.

## Example config (new style)

```yaml
server:
  rp_id: id.example.com
  rp_origins:
    - https://id.example.com
    - android:apk-key-hash:abc123def456
```

Or mixed (for gradual migration):

```yaml
server:
  rp_origin: https://id.example.com   # legacy — still works
  rp_origins:
    - android:apk-key-hash:abc123def456
```

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -786,8 +809,8 @@ func (c *Config) Validate() error {
 		return fmt.Errorf("rp_id is required")
 	}
 
-	if c.Server.RPOrigin == "" {
-		return fmt.Errorf("rp_origin is required")
+	if c.Server.RPOrigin == "" && len(c.Server.RPOrigins) == 0 {
```
[the remark this answers is not in the document]
Comment (leifj) [reply pkg/config/config.go]: Fixed in 3c34d23. `Validate()` now calls `len(c.Server.GetRPOrigins()) == 0` — since `GetRPOrigins()` already filters empty strings, a config with `rp_origins: [""]` is now correctly rejected. Also added a test case `"list of only empty strings is treated as empty"` to cover this explicitly.

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -81,14 +81,87 @@ func TestConfig_Validate_MissingRPOrigin(t *testing.T) {
...
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.cfg.GetRPOrigins()
+			if len(got) != len(tt.wantOrder) {
+				t.Fatalf("GetRPOrigins() = %v, want %v", got, tt.wantOrder)
+			}
+			for i, o := range got {
+				if o != tt.wantOrder[i] {
+					t.Errorf("GetRPOrigins()[%d] = %q, want %q", i, o, tt.wantOrder[i])
+				}
+			}
```
[the remark this answers is not in the document]
Comment (leifj) [reply pkg/config/config_test.go:174]: Fixed in 3c34d23. The test now explicitly checks `got != nil` when `wantOrder == nil` and returns early, so a non-nil empty slice would fail the assertion. The new `"list of only empty strings is treated as empty"` case also exercises the nil path with non-trivial input.

**Review by s-jairl — APPROVED** [review]
We probably want to update the example configuration files, but looks good besides that!

**Comment by leifj** [comment]
@copilot update the example files as suggested by @s-jairl

**Review by s-jairl — APPROVED** [review]
